packages/components/src/Textarea/Textarea.tsx

```diff
@@ -1,22 +1,58 @@
 import React from 'react';
-import { useStyles, system } from '@marigold/system';
+import { useStyles } from '@marigold/system';
+import { Exclamation, Required } from '@marigold/icons';
+import { ComponentProps } from '@marigold/types';
 
-export type TextareaProps = {
+import { ValidationMessage } from '../ValidationMessage';
+import { Label } from '../Label';
+import { Box } from '../Box';
+
+type TextareaProps = {
+  // how to check if theres a label and then htmlFor is required
   variant?: string;
-};
+  label?: string;
+  htmlFor?: string;
+  error?: string;
+  required?: boolean;
+} & ComponentProps<'textarea'>;
 
-export const Textarea = system<TextareaProps, 'textarea'>(
-  ({ variant = 'textarea', children, className, ...props }) => {
-    const classNames = useStyles(
-      {
-        variant: `form.${variant}`,
-      },
-      className
-    );
-    return (
-      <textarea className={classNames} {...props}>
+export const Textarea: React.FC<TextareaProps> = ({
+  variant = 'textarea',
+  htmlFor = 'textarea',
+  label,
+  error,
+  required = false,
+  children,
+  ...props
+}) => {
+  let errorColor = '#dd4142';
+  const errorClassName = useStyles({ color: errorColor });
+
+  return (
+    <Box display="block">
+      {label && (
+        <Label {...props} htmlFor={htmlFor}>
+          {label}
+          {(error || required) && (
+            <Required size={16} className={errorClassName} />
+          )}
+        </Label>
+      )}
+      <Box
+        as="textarea"
+        display="block"
+        border={error && `1px solid ${errorColor}`}
```

Comment:
use Outline to avoid the small "jump" of 1px on click (or add transparent 1px border if not clicked)